yarn.lock

```diff
@@ -653,1705 +412,1384 @@ __metadata:
-  conditions: os=linux & cpu=x64
+"@esbuild/sunos-x64@npm:0.21.5":
+  version: 0.21.5
+  resolution: "@esbuild/sunos-x64@npm:0.21.5"
+  conditions: os=sunos & cpu=x64
   languageName: node
   linkType: hard
 
-"@esbuild/netbsd-x64@npm:0.20.2":
-  version: 0.20.2
-  resolution: "@esbuild/netbsd-x64@npm:0.20.2"
-  conditions: os=netbsd & cpu=x64
+"@esbuild/sunos-x64@npm:0.24.0":
+  version: 0.24.0
+  resolution: "@esbuild/sunos-x64@npm:0.24.0"
+  conditions: os=sunos & cpu=x64
   languageName: node
   linkType: hard
 
-"@esbuild/openbsd-x64@npm:0.20.2":
-  version: 0.20.2
-  resolution: "@esbuild/openbsd-x64@npm:0.20.2"
-  conditions: os=openbsd & cpu=x64
+"@esbuild/win32-arm64@npm:0.21.5":
+  version: 0.21.5
+  resolution: "@esbuild/win32-arm64@npm:0.21.5"
+  conditions: os=win32 & cpu=arm64
   languageName: node
   linkType: hard
 
-"@esbuild/sunos-x64@npm:0.20.2":
-  version: 0.20.2
-  resolution: "@esbuild/sunos-x64@npm:0.20.2"
-  conditions: os=sunos & cpu=x64
+"@esbuild/win32-arm64@npm:0.24.0":
+  version: 0.24.0
+  resolution: "@esbuild/win32-arm64@npm:0.24.0"
+  conditions: os=win32 & cpu=arm64
   languageName: node
   linkType: hard
 
-"@esbuild/win32-arm64@npm:0.20.2":
-  version: 0.20.2
-  resolution: "@esbuild/win32-arm64@npm:0.20.2"
-  conditions: os=win32 & cpu=arm64
+"@esbuild/win32-ia32@npm:0.21.5":
+  version: 0.21.5
+  resolution: "@esbuild/win32-ia32@npm:0.21.5"
+  conditions: os=win32 & cpu=ia32
   languageName: node
   linkType: hard
 
-"@esbuild/win32-ia32@npm:0.20.2":
-  version: 0.20.2
-  resolution: "@esbuild/win32-ia32@npm:0.20.2"
+"@esbuild/win32-ia32@npm:0.24.0":
+  version: 0.24.0
+  resolution: "@esbuild/win32-ia32@npm:0.24.0"
   conditions: os=win32 & cpu=ia32
   languageName: node
   linkType: hard
 
-"@esbuild/win32-x64@npm:0.20.2":
-  version: 0.20.2
-  resolution: "@esbuild/win32-x64@npm:0.20.2"
+"@esbuild/win32-x64@npm:0.21.5":
+  version: 0.21.5
+  resolution: "@esbuild/win32-x64@npm:0.21.5"
   conditions: os=win32 & cpu=x64
   languageName: node
   linkType: hard
 
-"@faker-js/faker@npm:^8.3.1":
-  version: 8.3.1
-  resolution: "@faker-js/faker@npm:8.3.1"
-  checksum: 10/66b25cb82af907ff127767b54a5d6e95919c8051cd2de9131a6f5dd45dc539ebfbafad66e590e603e347e4c411f0f83c28e70043024af66faebba49d9868c263
+"@esbuild/win32-x64@npm:0.24.0":
+  version: 0.24.0
+  resolution: "@esbuild/win32-x64@npm:0.24.0"
+  conditions: os=win32 & cpu=x64
   languageName: node
   linkType: hard
 
-"@fal-works/esbuild-plugin-global-externals@npm:^2.1.2":
-  version: 2.1.2
-  resolution: "@fal-works/esbuild-plugin-global-externals@npm:2.1.2"
-  checksum: 10/fd68714cccfbd33a8ec31d11ac7c6373100a5e1b8e31941a45c723c802feccb0a00dde946f55cc91d58bff77d405adc2064b22f0faf5ee165968965e5da758a1
+"@faker-js/faker@npm:^8.3.1":
+  version: 8.4.1
+  resolution: "@faker-js/faker@npm:8.4.1"
+  checksum: 10/5983c2ea64f26055ad6648de748878e11ebe2fb751e3c7435ae141cdffabc2dccfe4c4f49da69a3d2add71e21b415c683ac5fba196fab0d5ed6779fbec436c80
   languageName: node
```

Comment:
> Odd, `yarn tsc` passes on main, but not on this branch. I'm wondering whether `sharp-ico` introduces a faker version somewhere? I tried upgrading to a more recent version of typescript and the error still persists. Specifying where types are stored also doesn't help.

@simonv3 it would seem it does going by the changes in `yarn.lock` here. Let me take a look at the library's dependencies to see if we can get one that depends on `main`'s version of `faker`.

Reply:
I tried to go by the [versions here](https://www.npmjs.com/package/sharp-ico?activeTab=versions) and tried `yarn install` for all of them. `yarn.lock` didn't change the `faker` version back to what it is on `main` so it looks like the newer version was the original dependency for `sharp-ico`.

Reply:
@lenikadali does freezing it prevent the type error?

Reply:
No unfortunately not.

This is what I see when I run `yarn tsc`:

```bash
$ yarn tsc
error TS2688: Cannot find type definition file for 'faker'.
  The file is in the program because:
    Entry point for implicit type library 'faker'


Found 1 error.
```

The other thing we could try is cherry-pick the commits onto `main` and see if `yarn tsc` passes (with the freeze).

Reply:
weird because the dependency tree of sharp-ico does not include faker:

```
└─ sharp-ico@0.1.5
   ├─ @types/sharp@0.30.5
   │  └─ @types/node@22.10.2
   │     └─ undici-types@6.20.0
   ├─ decode-ico@0.4.1
   │  ├─ to-data-view@1.1.0
   │  ├─ decode-bmp@0.2.1
   │  │  ├─ to-data-view@1.1.0
   │  │  └─ @canvas/image-data@1.0.0
   │  └─ @canvas/image-data@1.0.0
   ├─ ico-endec@0.1.6
   └─ sharp@0.33.5
      ├─ detect-libc@2.0.3
      ├─ @img/sharp-wasm32@0.33.5
      │  └─ @emnapi/runtime@1.3.1
      │     └─ tslib@2.8.1
      ├─ @img/sharp-linux-arm@0.33.5
      │  └─ @img/sharp-libvips-linux-arm@1.0.5
      ├─ color@4.2.3
      │  ├─ color-convert@2.0.1
      │  │  └─ color-name@1.1.4
      │  └─ color-string@1.9.1
      │     ├─ simple-swizzle@0.2.2
      │     │  └─ is-arrayish@0.3.2
      │     └─ color-name@1.1.4
      ├─ semver@7.6.3
      ├─ @img/sharp-linux-x64@0.33.5
      │  └─ @img/sharp-libvips-linux-x64@1.0.4
      ├─ @img/sharp-win32-x64@0.33.5
      ├─ @img/sharp-darwin-x64@0.33.5
      │  └─ @img/sharp-libvips-darwin-x64@1.0.4
      ├─ @img/sharp-win32-ia32@0.33.5
      ├─ @img/sharp-linux-arm64@0.33.5
      │  └─ @img/sharp-libvips-linux-arm64@1.0.4
      ├─ @img/sharp-linux-s390x@0.33.5
      │  └─ @img/sharp-libvips-linux-s390x@1.0.4
      ├─ @img/sharp-darwin-arm64@0.33.5
      │  └─ @img/sharp-libvips-darwin-arm64@1.0.4
      ├─ @img/sharp-linuxmusl-x64@0.33.5
      │  └─ @img/sharp-libvips-linuxmusl-x64@1.0.4
      ├─ @img/sharp-linuxmusl-arm64@0.33.5
      │  └─ @img/sharp-libvips-linuxmusl-arm64@1.0.4
      ├─ @img/sharp-libvips-linux-x64@1.0.4
      ├─ @img/sharp-libvips-linux-arm@1.0.5
      ├─ @img/sharp-libvips-darwin-x64@1.0.4
      ├─ @img/sharp-libvips-linux-s390x@1.0.4
      ├─ @img/sharp-libvips-linux-arm64@1.0.4
      ├─ @img/sharp-libvips-darwin-arm64@1.0.4
      ├─ @img/sharp-libvips-linuxmusl-x64@1.0.4
      └─ @img/sharp-libvips-linuxmusl-arm64@1.0.4
```

Reply:
Okay, I think something else happened here. When I go to the main branch, create a new branch for this, and install `sharp-ico` from scratch, I don't get the type error. 

@lenikadali I wonder if starting from scratch here would be a good idea? Basically just a blank branch, install `sharp-ico` and then add the lines changed. I'd do the copy paste job but want to make sure you get credit!

Reply:
Hi @simonv3 that's a good idea. New PR is [here](https://github.com/funmusicplace/mirlo/pull/937)

If the pipeline passes for that one, this PR can be closed :crossed_fingers: